### PR TITLE
web/api: add alertmanager api

### PR DESCRIPTION
**Pull request by brancz**
Similar to the Targets API that was recently added in #2234. We would like to test whether a Prometheus instance has been setup correctly and is properly discovering Alertmanagers. The particular case is for the [prometheus-operator](https://github.com/coreos/promethesu-operator)'s end-to-end tests. Ultimately this as well as the Targets API introduced in #2234 should be exposing dropped targets as well, which would help a lot when debugging a configuration, for that however, some more ground work and refactoring in the retrievel/discovery mechanisms need to be made. That is also the reason that some of this code is very similar to the Targets API, as I did not want to introduce a premature abstraction that is too tightly coupled and won't allow those changes to be easily made.

If possible when releasing these two APIs I would like to release them in beta status as due to the above reasons they are likely to be subject to change in the future.

@alexsomesan @brian-brazil @fabxc

### CODE REVIEW

File: web/api/v1/api.go [hunk 1]
```diff
@@ -375,6 +383,30 @@ func (api *API) targets(r *http.Request) (interface{}, *apiError) {
...
+	URL string `json:"url"`
+}
+
+func (api *API) alertmanagers(r *http.Request) (interface{}, *apiError) {
+	amTargets := api.alertmanagerRetriever.AlertmanagerTargets()
+	res := make([]*AlertmanagerTarget, len(amTargets))
+
+	for i, t := range amTargets {
+		res[i] = &AlertmanagerTarget{
+			DiscoveredLabels: t.DiscoveredLabels(),
```
Comment (brian-brazil) [inline web/api/v1/api.go]: We don't use labels like we do for normal targets, so I don't think we should expose them to users to avoid confusion.

Comment (brancz) [inline web/api/v1/api.go]: Do you think we should just return a list of URLs then?

Comment (brian-brazil) [inline web/api/v1/api.go]: Yes, but do it as a map to allow for future expansion.

**Comment by brancz** [comment]
Fixed. I think we should do the same for the targets API then. So we have a struct that contains `activeTargets` and `droppedTargets`, currently the result is list of all currently active targets. Wdyt @brian-brazil @fabxc ? That will allow us to have a non breaking change when we do add dropped targets/alertmanagers as it will just add a field to the struct.

**Comment by brian-brazil** [comment]
Consistency is good.

**Comment by fabxc** [comment]
Yea I think this one should be an object list rather than a string list.

`{
    url: "http://bla.bla/api/v1/alerts"
}`

We might want to extend it by fields like `lastError`  later, similar to targets.

**Comment by brancz** [comment]
Adapted. And also changed the response for the targets API. Let me know if you want that in a separate PR, for now I pushed it onto this one. PTAL @brian-brazil @fabxc

File: web/api/v1/api.go [hunk 2]
```diff
@@ -344,16 +351,20 @@ type Target struct {
 	// Any labels that are added to this target and its metrics.
 	Labels model.LabelSet `json:"labels"`
 
-	ScrapeUrl string `json:"scrapeUrl"`
+	ScrapeURL string `json:"scrapeURL"`
```
Comment (fabxc) [inline web/api/v1/api.go]: I think in general in JSON it would stay `scrapeUrl`, no? 
Are we doing the full capsing elsewhere?

Comment (brancz) [inline web/api/v1/api.go]: fixed

**Pull request merged**